File: filesystem/src/grammar/tests/validation_data/rs/validator_erroneous_code.rs
// Rust validation file with syntax errors to test parser robustness

// Error: Missing semicolon
fn missing_semicolon() -> i32 {
    let x = 5
    x + 1
}

// Error: Using a mutable variable without declaring it as mutable
fn mutability_error() {
    let x = 5;
    x = 10; // Error: cannot assign twice to immutable variable
}

// Error: Mismatched types
fn type_mismatch() -> i32 {
    "not an integer" // Error: expected i32, found &str
}

// Error: Undefined variable
fn undefined_variable() {
    println!("{}", nonexistent_variable);
}

// Error: Missing return type
fn missing_return_type() {
    42 // Error: missing return type for function
}

// Error: Extra token
struct ExtraToken {
    field: i32,
}; // Error: extra semicolon

// Error: Mismatched brackets
fn mismatched_brackets() {
    let values = vec![1, 2, 3); // Error: mismatched brackets
}

// Error: Incorrect lifetime usage
fn incorrect_lifetime<'a, 'b>(x: &'a str, y: &'b str) -> &'c str { // Error: undeclared lifetime
    x
}

// Error: Broken match pattern
fn broken_match(option: Option<i32>) -> i32 {
    match option {
        Some => 42, // Error: expected pattern, found identifier
        None => 0,
    }
}

// Error: Invalid struct field
struct InvalidField {
    let value: i32, // Error: expected identifier, found keyword
}

// Error: Comparing incomparable types
fn compare_error() -> bool {
    "string" == 42 // Error: can't compare &str and i32
}

// Error: Invalid method call
fn invalid_method() {
    let x = 5;
    x.nonexistent_method(); // Error: no method named `nonexistent_method`
}

// Error: Private field access
mod privacy_error {
    struct Private {
        private_field: i32,
    }
    
    fn access_outside() {
        let p = Private { private_field: 42 };
    }
}

fn access_error() {
    let p = privacy_error::Private { private_field: 42 }; // Error: private field
}

// Error: Missing trait implementation
trait Required {
    fn required_method(&self);
}

struct MissingImpl;

impl Required for MissingImpl {
    // Error: missing implementation for required method
}

// Error: Out of bounds array access with const
fn array_bounds() {
    let arr = [1, 2, 3];
    let value = arr[5]; // Error: index out of bounds
}

// Error: Invalid loop label
fn invalid_label() {
    'outer: for i in 0..5 {
        for j in 0..5 {
            break 'nonexistent; // Error: use of undeclared label
        }
    }
}

// Error: Type parameter used in function without being declared
fn undefined_type_param(value: T) -> T { // Error: use of undeclared type name
    value
}

// Error: Multiple incompatible trait bounds
fn incompatible_bounds<T: Clone + Copy + AsRef<str>>(value: T) { // Error: incompatible bounds
    // Can't be both str and have copy semantics
}

// Error: Invalid use of mut in parameter
fn invalid_mut(mut self) { // Error: invalid `mut` in function parameter
    // Only valid in impl methods
}

// Error: Reserved keyword as identifier
fn let() { // Error: expected identifier, found keyword
    println!("Can't use keywords as identifiers");
}

// Error: Invalid binary operation
fn invalid_operation() {
    let result = "string" + "concatenation"; // Error: cannot add two &str values
}

// Error: Missing generic parameter
struct GenericStruct<T> {
    field: T,
}

fn generic_error() {
    let value = GenericStruct { field: 42 }; // Error: missing type parameters
}

// Error: Multiple definition errors in one function
fn multiple_errors() {
    let x: i32 = "string"; // Type mismatch
    let y = z; // Undefined variable
    if (x = 5) { // Assignment in condition
        break; // Break outside of loop
    }
} 

File: filesystem/src/grammar/tests/validation_data/cpp/validator_erroneous_code.cpp
// C++ validation file with syntax errors to test parser robustness

#include <iostream>
#include <vector>
#include <string>
#include <memory>

// Error: Extra token
class ExtraToken {
public:
    int value;
};; // Extra semicolon

// Error: Missing semicolon
class MissingSemicolon {
public:
    int value
}; // Missing semicolon after 'value'

// Error: Mismatched brackets
void mismatched_brackets() {
    std::vector<int> values = {1, 2, 3];  // Mismatched brackets
}

// Error: Mismatched parentheses
void mismatched_parentheses(int x, int y {  // Missing closing parenthesis
    return x + y;
}

// Error: Undefined variable
void undefined_variable() {
    int x = 10;
    std::cout << y << std::endl;  // 'y' is not defined
}

// Error: Type mismatch
void type_mismatch() {
    int x = "string";  // Cannot assign string to int
    std::string s = 42;  // Cannot assign int to string
}

// Error: Duplicate variable declaration
void duplicate_variable() {
    int x = 10;
    int x = 20;  // Redeclaration of 'x'
}

// Error: Wrong function call syntax
void wrong_function_call() {
    std::cout.println("Hello");  // No 'println' method in cout
}

// Error: Using class template without template arguments
void missing_template_args() {
    std::vector v;  // Missing template arguments for vector
}

// Error: Wrong template syntax
template <class T, class> class WrongTemplate {  // Missing template parameter name
    T value;
};

// Error: Access control conflict
class AccessConflict {
public:
private:
public:  // Duplicate access specifier
    int value;
};

// Error: Multiple inheritance ambiguity
class Base1 {
public:
    int value = 1;
};

class Base2 {
public:
    int value = 2;
};

class Derived : public Base1, public Base2 {  // Diamond problem
public:
    void printValue() {
        std::cout << value;  // Ambiguous: Base1::value or Base2::value?
    }
};

// Error: Incorrect lambda syntax
auto lambda = [x, y]() {  // x and y not captured or declared
    return x + y;
};

// Error: Missing template keyword
void template_error() {
    std::vector<int> v;
    auto it = v.begin();
    std::vector<int>::iterator::value_type val = *it;  // Missing 'typename'
}

// Error: Duplicate template parameter
template <typename T, typename T>  // Duplicate 'T'
class DuplicateParam {
    T value;
};

// Error: Missing operator in expression
void missing_operator() {
    int x = 5;
    int y = 10;
    int z = x y;  // Missing operator between x and y
}

// Error: Misuse of scope resolution operator
void scope_error() {
    std::cout::endl;  // Incorrect use of scope resolution
}

// Error: Declaring a variable with 'void' type
void type_void_error() {
    void x = 10;  // Cannot declare variable of type 'void'
}

// Error: Multiple unrelated errors in one function
void multiple_errors() {
    int x = "string";  // Type mismatch
    y = 20;  // Undefined variable
    std::vector<> v;  // Missing template argument
    v.non_existent_method();  // Non-existent method
}

// Error: Break outside of loop
void break_outside_loop() {
    if (true) {
        break;  // Break outside of loop
    }
}

// Error: Using non-static member without object
class NonStaticMember {
public:
    void method() {}
};

void call_without_object() {
    NonStaticMember::method();  // Calling non-static method without object
}

// Error: Missing return statement
int missing_return() {
    int x = 10;
    // Missing return statement
}

// Error: Wrong inheritance syntax
class Base {
public:
    virtual void foo() {}
};

class WrongInheritance : Base {  // Missing 'public', 'protected', or 'private'
public:
    void foo() override {}
};

// Error: Invalid delete
void invalid_delete() {
    int x = 10;
    delete x;  // Cannot delete non-pointer
}

// Error: Comparison between pointer and integer
void pointer_integer_comparison(int* ptr) {
    if (ptr == 5) {  // Comparing pointer to integer
        std::cout << "Equal" << std::endl;
    }
}

// Error: Invalid struct initialization
struct Point {
    int x;
    int y;
};

void invalid_initialization() {
    Point p = {1, 2, 3};  // Too many initializers
}

// Error: Incomplete enum value assignment
enum Color {
    RED = ,  // Missing value
    GREEN,
    BLUE
}; 